src/mon/PGMap.cc

```diff
@@ -2056,7 +2056,7 @@ void PGMap::get_filtered_pg_stats(uint64_t state, int64_t poolid, int64_t osdid,
       continue;
     if ((osdid >= 0) && !(i->second.is_acting_osd(osdid,primary)))
       continue;
-    if (!(i->second.state & state))
+    if (state && !(i->second.state & state))
```

Comment:
So if you want to see all states including i->second.state == 0 you don't specify any states to pg-ls?  Because specifying "all" gets you pgs that have any state bits set.

Reply:
The problem is that i->second.state == 0 for the "unknown" case, so this statement is always false (even if, say, the passed-in state has all bits set).

Reply:
oh, right!